node/process.ts

```diff
@@ -258,6 +262,13 @@ class Process extends EventEmitter {
   constructor() {
     super();
 
+    globalThis.addEventListener("beforeunload", (e) => {
+      super.emit("beforeExit", process.exitCode || 0);
+      if (core.eventLoopHasMoreWork()) {
```

Comment:
This event listener should call into `processTicksAndRejections` from next_tick.ts before checking if there's more work. That would be enough to fix this issue, with no need to change Deno's event loop.

Reply:
Do you mean like this:

```js
    globalThis.addEventListener("beforeunload", (e) => {
      super.emit("beforeExit", process.exitCode || 0);
      processTicksAndRejections();
      if (core.eventLoopHasMoreWork()) {
        e.preventDefault();
      }
    });
```

If so, that doesn't seem to get `deno test -A --unstable node/_tools/test.ts -- parallel/test-process-beforeexit.js` passing.

Reply:
It doesn't pass with Deno 1.23.4, but for me it's working with canary.

Reply:
Ah, right - because https://github.com/denoland/deno/pull/14830 hasn't been released yet 🤦 . It's working for me with latest main.

Reply:
Thanks for figuring it out @andreubotella!